pkg/receive/handler.go

```diff
@@ -464,20 +465,57 @@ func (h *Handler) replicate(ctx context.Context, tenant string, wreq *prompb.Wri
 	}
 	h.mtx.RUnlock()
 
-	err := h.parallelizeRequests(ctx, tenant, replicas, wreqs)
-	if errs, ok := err.(terrors.MultiError); ok {
-		if uint64(countCause(errs, isNotReady)) >= (h.options.ReplicationFactor+1)/2 {
-			return tsdb.ErrNotReady
-		}
-		if uint64(countCause(errs, isConflict)) >= (h.options.ReplicationFactor+1)/2 {
-			return errors.Wrap(conflictErr, "did not meet replication threshold")
-		}
-		if uint64(len(errs)) >= (h.options.ReplicationFactor+1)/2 {
-			return errors.Wrap(err, "did not meet replication threshold")
+	ctx, cancel := context.WithTimeout(ctx, h.options.ForwardTimeout)
+	success := uint64(0)
+
+	n, ec := h.parallelizeRequests(ctx, tenant, replicas, wreqs)
+	defer func() {
+		go func() {
+			for {
+				err, more := <-ec
```

Comment:
```
defer cancel()
for err := range <-ec {
```
Would do the work I think.

Reply:
Ok because you relied on err != nil on caller side.... and you forgot about it here, I would really recommend my suggestion in comment above =D

Reply:
And yes, I understand this is how we know if it was success or not.. but maybe we can come up with cleaner API

Reply:
The problem is that we want to drain the channel, and stop when it's actually empty and closed, so we need to have the `more` returned.

Any suggestions for a better API? The problem is that we need to know in advance how many potential results we would be getting from the channel.

Reply:
But `for ... <-ec` has exactly the same semantics, no?

Reply:
It will stop iterating when it's closed. If empty `err, more := <-ec` will not unblock either, it will block